python/cudf/cudf/testing/testing.py

```diff
@@ -201,31 +206,49 @@ def assert_column_equal(
     ):
         left = left.astype(left.categories.dtype)
         right = right.astype(right.categories.dtype)
-
     columns_equal = False
-    try:
-        columns_equal = (
-            (
-                cp.all(left.isnull().values == right.isnull().values)
-                and cp.allclose(
+    if left.size == right.size == 0:
+        columns_equal = True
+    elif not (
+        (is_string_dtype(left) and is_numeric_dtype(right))
+        or (is_numeric_dtype(left) and is_string_dtype(right))
+    ):
+        try:
+            # nulls must be in the same places for all dtypes
+            nulls_equal = cp.all(left.isnull().values == right.isnull().values)
```

Comment:
Do you need to access the values here? I think you can just rely on `__cuda_array_interface__`.

Reply:
It looks like `cp.all` only plays nice with `cupy` arrays:

```python
>>> cupy.all(
...     cudf.core.column.as_column([True, False, True])
... )
TypeError: 'a' must be a cupy.ndarray object, not <class 'cudf.core.column.numerical.NumericalColumn'>.
```

Reply:
I see, that's kind of annoying that it doesn't try to convert internally.